plugins/woocommerce/client/admin/client/core-profiler/pages/IntroOptIn.tsx

```diff
@@ -126,7 +127,14 @@ export const IntroOptIn = ( {
 							},
 						} ) }
 						checked={ iOptInDataSharing }
-						onChange={ setIsOptInDataSharing }
+						onChange={ ( value ) => {
+							recordEvent( 'woocommerce_allow_tracking_toggled', {
+								previous_value: ! value,
+								new_value: value,
+								context: 'core-profiler',
+							} );
```

Comment:
>  I do see the tracking pixel sent every time I toggle the value of the checkbox (not when the value is saved):

Thats a fair point. I updated everything now so it's only sent when the value is actually saved (works for both, yes to no and no to yes).

Reply:
> This is expected to not track the toggle. I thought about trigger it o update option hook, however it will be triggered everywhere. So I consider this case using the CLI as manually changed. It will trigger the ones in WC_Trcks tho. I can investigate in further PRs how can we do to catch when the option is updated via CLI

Thanks for the clarification. I wasn't expecting to track toggling through WP CLI. As in theory you could also go to wp-admin/options.php or edit in the DB through many other ways, which I don't think we can track.

My main intention was to start onboarding with the value toggled off (as in the default scenario when starting onboarding the first time). It's possible to start onboarding at a later stage, so we need to confirm it works with previous states of the setting.